### DISCUSSION TITLE
Surface Effect Correction with Peridigm

### DISCUSSION THREAD

**Discussion by paulordutra** [peridigm/peridigm#376] [Q&A]
Hi everyone,

I'm working on implementing a surface effect correction scheme for BBPD, as outlined in Chapter 2 of the _Handbook of Peridynamics_. In this scheme, the micromodulus c is adjusted by a factor that accounts for the horizon volume of a particle within the main bulk and the neighborhood volumes of two particles in any bond. To do this, I need to access the NeighborhoodVolumeFieldId.

I attempted to use it as the VolumeFieldId, but it does not return the volumes as expected. When running a simple model, the force is too large and the software crashes. I suppose that it's computed after the force, an output, and, for this reason, the force is infinite (null volumes). 

How could I properly access or compute Neighborhood_Volume in this context?

Thanks in advance!

**Comment by JTHesse** [comment]
Take a look at [PeriLab.jl](https://github.com/PeriHub/PeriLab.jl). Surface Correction is implemented there.
